toonz/sources/tnztools/brushtool.cpp

```diff
@@ -1616,28 +1616,40 @@ void BrushTool::mouseMove(const TPointD &pos, const TMouseEvent &e) {
       setValue(prop, value);
     }
 
-  } locals = {this};
+    void addMinMaxSeparate(TDoublePairProperty &prop, double min, double max) {
+      if (min == 0.0 && max == 0.0) return;
+      const TDoublePairProperty::Range &range = prop.getRange();
 
-  switch (e.getModifiersMask()) {
-  /*--
-   * Altキー+マウス移動で、ブラシサイズ（Min/Maxとも）を変える（CtrlやShiftでは誤操作の恐れがある）
-   * --*/
-  case TMouseEvent::ALT_KEY: {
-    // User wants to alter the minimum brush size
-    const TPointD &diff = pos - m_mousePos;
-    double add          = (fabs(diff.x) > fabs(diff.y)) ? diff.x : diff.y;
+      TDoublePairProperty::Value value = prop.getValue();
+      value.first += min;
+      value.second += max;
+      if (value.first > value.second) value.first = value.second;
+      value.first  = tcrop(value.first, range.first, range.second);
+      value.second = tcrop(value.second, range.first, range.second);
 
-    locals.addMinMax(
-        TToonzImageP(getImage(false, 1)) ? m_rasThickness : m_thickness, add);
+      setValue(prop, value);
+    }
 
-    break;
-  }
+  } locals = {this};
 
-  default:
+  // if (e.isAltPressed() && !e.isCtrlPressed()) {
+  // const TPointD &diff = pos - m_mousePos;
+  // double add = (fabs(diff.x) > fabs(diff.y)) ? diff.x : diff.y;
+
+  // locals.addMinMax(
+  //  TToonzImageP(getImage(false, 1)) ? m_rasThickness : m_thickness, add);
+  //} else
+  if (e.isCtrlPressed() && e.isAltPressed()) {
+    const TPointD &diff = pos - m_mousePos;
+    double max          = diff.x / 2;
+    double min          = diff.y / 2;
+
+    locals.addMinMaxSeparate(
+        TToonzImageP(getImage(false, 1)) ? m_rasThickness : m_thickness, min,
+        max);
+  } else {
```

Comment:
I know it is the problem included in the original source and not of your PR, but could you please replace lines 1648-1649 by the following?:

`(m_targetType & TTool::ToonzImage) ? m_rasThickness : m_thickness, min, max);`

Because when the empty cell is selected, `getImage(false, 1)` returns 0 so `addMinMaxSeparate` will try to change `m_thickness` (= for vector brush) . But the current brush tool for creating new level is set according to the Preferences > Drawing > Default Level Type and it can be the raster brush.
So I think it would be better to use the target type for identifing the brush type (just like the same way done in line 1657 of this source).